src/main/java/spliterators/part1/exercise/RectangleSpliterator.java

```diff
@@ -33,31 +33,21 @@ private RectangleSpliterator(int[][] array, int startOuterInclusive, int endOute
 
     @Override
     public OfInt trySplit() {
-        final int outerLength = endOuterExclusive - startOuterInclusive;
-        final int totalLength;
-        if (outerLength < 1){
-            return null;
-        } else if (outerLength == 1){
-            totalLength = endInnerExclusive - startInnerInclusive;
-        } else{
-            totalLength = (innerLength - startInnerInclusive) + innerLength*(outerLength-2) + endInnerExclusive;
-        }
+        final int totalLength = (int) estimateSize();
         if (totalLength < 2){
             return null;
         }
         final RectangleSpliterator result;
         int lengthLeft = totalLength / 2;
         if (lengthLeft - (innerLength - startInnerInclusive) <= 0){     // outerLength == 1
-            result = new RectangleSpliterator(array,startOuterInclusive,startOuterInclusive + 1,startInnerInclusive,
-                    startInnerInclusive + lengthLeft);
+            result = new RectangleSpliterator(array,startOuterInclusive,startOuterInclusive + 1,startInnerInclusive,startInnerInclusive + lengthLeft);

```

Comment:
Try to use a single line with `new RectangleSpliterator` in this method.
I mean a single `new` operator in this method.